vllm/worker/multi_step_model_runner.py

```diff
@@ -486,27 +481,27 @@ def _update_sampling_metadata(self, sampling_metadata, num_seqs,
 
     def _advance_step(self, model_input: StatefulModelInput,
                       out: SamplerOutput) -> StatefulModelInput:
-        frozen_model_input = model_input.frozen_model_input
-        assert frozen_model_input is not None
-        assert frozen_model_input.attn_metadata is not None
+        if self.attn_backend.get_name() not in MULTI_STEP_ATTENTION_BACKENDS:
+            raise ValueError(
+                f"Multi-step not supported for attention backend: "
+                f"{self.attn_backend.get_name()}. Set VLLM_ATTENTION_BACKEND "
+                "to 'flash-attn' or 'flashinfer'.")
```

Comment:
List names from the MULTI_STEP_ATTENTION_BACKENDS

Reply:
3112e85